pkg/virt-controller/services/template.go

```diff
@@ -468,7 +473,40 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, i
 		containers = append(
 			containers,
 			newSidecarContainerRenderer(
-				sidecarContainerName(i), vmi, sidecarResources(vmi), requestedHookSidecar, userId).Render(requestedHookSidecar.Command))
+				sidecarContainerName(i), requestedHookSidecar.Image, requestedHookSidecar.ImagePullPolicy, vmi, sidecarResources(vmi), userId,
+				WithVolumeMounts(sidecarHookVolumeMount()),
+				WithArgs(requestedHookSidecar.Args),
+			).Render(requestedHookSidecar.Command))
+	}
+
+	toShareProcessNsInPod := false
+	if vmi.IsCPUDedicated() {
+		resourceOptions := []ResourceRendererOption{
+			WithEphemeralStorageRequest(),
+			WithMemoryOverhead(vmi.Spec.Domain.Resources, getVcpusOverhead(vmi)),
+			WithCPUPinning(vmi.Spec.Domain.CPU),
+		}
+
+		if cgroups.IsCgroup2UnifiedMode() && !vmi.IsIsolated() {
```

Comment:
I agree.
The problem here is that while virt-launcher prepares the pod there is no way of knowing to which node it would be scheduled later. If we can't assume that nodes have the same cgroup configuration, I'm not sure how we can get around the +1 core problem for v1 environments. We can solve that by always allocating one extra CPU, but I wonder if there's a better idea to handle that.